addons/product/models/product_attribute.py

```diff
@@ -22,8 +22,30 @@ class ProductAttribute(models.Model):
         ('always', 'Always'),
         ('dynamic', 'Only when the product is added to a sales order')],
         default='always',
-        string="Create Variants",
-        help="Check this if you want to create multiple variants for this attribute.", required=True)
+        states={'used_on_product': [('readonly', True)]},
+        string="Variants Creation Mode",
+        help="This cannot be changed if the attribute is used on at least one product.", required=True)
+
+    state = fields.Selection(compute='_compute_state',
```

Comment:
Maybe `Boolean` field instead of selection (Saying based on purpose)
Eg, `used_on_product` (Which should be stored and depends on attribute_line_ids)

What do you think on this ?

Reply:
Yep use a boolean : `has_used_on_product`

Reply:
> Yep use a boolean : `has_used_on_product`

or `is_used_on_product`

Reply:
The goal here is to take advantage of `states=` on the above field to set it `readonly` when appropriate.